pkg/materialize/generic.go

```diff
@@ -79,53 +80,54 @@ func (b *Builder) resize(name, size string) error {
 	return b.exec(q)
 }
 
-func (b *Builder) alter(name string, options map[string]interface{}, isSecret, validate bool) error {
-	var setClauses []string
-	for option, val := range options {
+func (b *Builder) alter(name string, setOptions map[string]interface{}, resetOptions []string, isSecret, validate bool) error {
+	var clauses []string
+	for option, val := range setOptions {
 		var setValue string
 		switch v := val.(type) {
 		case ValueSecretStruct:
 			if v.Text != "" {
-				setValue = fmt.Sprintf("%s", QuoteString(v.Text))
+				setValue = QuoteString(v.Text)
 			} else if v.Secret.Name != "" {
-				setValue = fmt.Sprintf("SECRET %s", v.Secret.QualifiedName())
+				setValue = "SECRET " + v.Secret.QualifiedName()
 			}
 		case IdentifierSchemaStruct:
 			prefix := ""
-			if isSecret {
+			if isSecret || option == "SASL PASSWORD" || option == "SSL KEY" {
 				prefix = "SECRET "
 			}
-			setValue = fmt.Sprintf("%s%s", prefix, v.QualifiedName())
+			setValue = prefix + v.QualifiedName()
 		case string:
-			setValue = fmt.Sprintf("%s", QuoteString(v))
+			setValue = QuoteString(v)
 		case int:
 			setValue = fmt.Sprintf("%d", v)
 		case []string:
-			if len(v) > 0 {
-				setValue = fmt.Sprintf("[%s]", "'"+strings.Join(v, "', '")+"'")
-			} else {
-				setValue = "[]"
-			}
+			setValue = fmt.Sprintf("[%s]", "'"+strings.Join(v, "', '")+"'")
+		case RawSQL:
+			setValue = string(v)
 		default:
 			return fmt.Errorf("unsupported value type for option %s: %T", option, val)
 		}
 
 		if setValue == "" {
-			return fmt.Errorf("no valid value provided for option %s: %T", option, val)
+			return fmt.Errorf("no valid value provided for option %s", option)
 		}
 
-		setClause := fmt.Sprintf("%s = %s", option, setValue)
-		setClauses = append(setClauses, setClause)
+		clauses = append(clauses, fmt.Sprintf("SET (%s = %s)", option, setValue))
+	}
+
+	for _, option := range resetOptions {
+		clauses = append(clauses, fmt.Sprintf("RESET (%s)", option))
```

Comment:
If we try to quote this it will fail with:

```
ERROR:  Expected one of ACCESS or ASSUME or AVAILABILITY or AWS or BROKER or BROKERS or DATABASE or ENDPOINT or HOST or PASSWORD or PORT or PROGRESS or REGION or ROLE or SASL or SECRET or SECURITY or SERVICE or SESSION or SSH or SSL or URL or USER or USERNAME, found identifier "HOST"
LINE 1: alter connection ssh_conn reset ("HOST");
```